geode-core/src/distributedTest/java/org/apache/geode/internal/statistics/StatisticsDistributedTest.java

```diff
@@ -490,6 +490,53 @@ public void testPubAndSubCustomStats() throws Exception {
         combinedUpdateEvents == combinedPuts);
   }
 
+  @Test
+  public void testExistenceOfVMStats() {
+    String regionName = "region_1";
+    VM vm = getHost(0).getVM(0);
+
+    vm.invoke(() -> {
+      Properties props = new Properties();
+      props.setProperty(STATISTIC_SAMPLING_ENABLED, "true");
+      props.setProperty(STATISTIC_SAMPLE_RATE, "1000");
+
+      InternalDistributedSystem system = getSystem(props);
+
+      // assert that sampler is working as expected
+      GemFireStatSampler sampler = system.getStatSampler();
+      assertTrue(sampler.isSamplingEnabled());
+      assertTrue(sampler.isAlive());
+
+      await("awaiting SampleCollector to exist")
+          .until(() -> sampler.getSampleCollector() != null);
+
+      SampleCollector sampleCollector = sampler.getSampleCollector();
+      assertNotNull(sampleCollector);
+
+      StatisticsType VMStatsType = getSystem().findType("VMStats");
+      Statistics vmStats = getSystem().findStatisticsByType(VMStatsType)[0];
+
+      try {
+        vmStats.get("pendingFinalization");
+        vmStats.get("loadedClasses");
+        vmStats.get("unloadedClasses");
+        vmStats.get("daemonThreads");
+        vmStats.get("peakThreads");
+        vmStats.get("threads");
+        vmStats.get("threadStarts");
+        vmStats.get("cpus");
+        vmStats.get("freeMemory");
+        vmStats.get("totalMemory");
+        vmStats.get("maxMemory");
+        vmStats.get("processCpuTime");
+        vmStats.get("fdsOpen");
+        vmStats.get("fdLimit");
+      } catch (IllegalArgumentException e) {
+        fail("Stats do not exist: " + e.getMessage());
```

Comment:
I would love to see us shy away from the this style of testing where we catch exceptions and then `fail`. I prefer that we either let the exception be thrown, and that in itself fails the tests or AssertJ's `assertThatNoException().isThrownBy(() -> {something()})`.

Reply:
thank you! I was trying to use `assertThatThrownBy(() -> something).doesNotThrow()` and it was failing when the stat exists/no exception is thrown